### DISCUSSION TITLE
Looping array in automation mapping issue and hangs in test mode

### DISCUSSION THREAD

**Discussion by mattiason** [Budibase/budibase#6255] [Help]
Hi

I am trying to loop an array in automation to perform a search but I cant make it work. It only hangs when I am doing the test 

I using a rest API get the external REST connector 

1. Output form the external data source conntector
<img width="359" alt="Screenshot 2022-06-09 at 20 43 27" src="https://user-images.githubusercontent.com/45860668/172921836-032ca764-ef3d-4b16-9b19-0b545def2462.png">

2. Mapping in the looping
<img width="568" alt="Screenshot 2022-06-09 at 20 45 05" src="https://user-images.githubusercontent.com/45860668/172921959-e8d0f296-d7c5-48c9-b520-0cc7c67bf341.png">

3. Mapping for the filter 
<img width="1139" alt="Screenshot 2022-06-09 at 20 45 17" src="https://user-images.githubusercontent.com/45860668/172922498-47f8c60c-18bf-49d2-87ea-29e6286240a5.png">

4 When I press test the following is happening it is just getting stuck in the test. If remove the looping the test works again
<img width="906" alt="Screenshot 2022-06-09 at 20 49 49" src="https://user-images.githubusercontent.com/45860668/172922245-5e9e61a6-8e12-44f4-b078-da5196738594.png">

**Comment by melohagan** [comment]
Hi @mattiason 

This is a bug which I have a PR out for: https://github.com/Budibase/budibase/pull/6246
